.github/workflows/tests.yml

```diff
@@ -54,7 +56,7 @@ jobs:
 
     # Note matrix-job outputs are kind of weird. Last-to-run job that sets a non-empty value wins.
     outputs:
-      did-coverage: ${{ ( steps.process-coverage.conclusion == 'success' && steps.upload-artifacts.conclusion == 'success' ) && 'true' || '' }}
+      did-coverage: ${{ ( steps.run-tests.conclusion != 'cancelled' && steps.process-coverage.conclusion == 'success' && steps.upload-artifacts.conclusion == 'success' ) && 'true' || '' }}
```

Comment:
Should we even try to run `process-coverage` if `run-tests` is cancelled? 🤔 Though I guess if it runs that's an easy way to determine if coverage was enabled and I'm not sure if `matrix.script == 'test-coverage'` is accessible here.

Reply:
For manual analysis, it probably doesn't hurt to save whatever coverage data we did manage to generate. But, particularly for trunk runs, it'll throw things off if that partial data gets uploaded.

The check here is now threefold: (1) we finished running the actual tests, (2) we successfully processed the coverage data, and (3) we successfully uploaded the artifact. If 2 or 3 is false, the upload job won't even be able to run successfully because there will be no data. If 1 is false, it's partial data that we don't want to upload.